### DISCUSSION TITLE
Setting up multiple Mediamtx servers and configure it with Network Load Balancer

### DISCUSSION THREAD

**Discussion by GPUbrainStorm** [bluenviron/mediamtx#3377] [Questions] [answered]
### Question

Hello all,

I'm trying to setup an AWS Network Load Balancer with multiple mediamtx servers on AWS to stream using WebRTC (WHIP/WHEP) but I couldnt figure out how to configure the load balancer and the mediamtx and not sure what's going wrong!
I setup the NLB to listen to port 8889 and route the traffic to the same port on the server but for some reason it's not working even though the health check is returning healthy. What are the adjustments am I supposed to do? or some source that I can look at to understand how it is supposed to be in order to work?
I'm still a bit new to the whole thing, AWS and mediamtx so sorry if I said some nonsense in my question please feel free to correct me if am getting something wrong.

I'm using v1.8.2

Thank you,

**Comment by GPUbrainStorm** [comment, marked as answer]
I was able to figure it out, if anyone need some assist feel free to reach out.
Thanks.

**Reply by danidroidvx** [reply]
hi, did you have to add manual the listeners range of ports ? or your solution just worked opening one port?

**Reply by GPUbrainStorm** [reply]
I used one port only. I configured the load balancer to redirect all the traffic on port 443 to the MediaMTX server port which is 8889 by default for WebRTC. All servers were listening to the same port so no matter what server is selected by the load balancer it will be fine.

**Comment by caleyg** [comment]
Hello, we're curious what you did on this ty

**Reply by GPUbrainStorm** [reply]
For WebRTC I needed to first point the domain to a server, generate a certificate and add it to the MediaMTX config then copied it too all other servers. After that I configured the load balancer to redirect all the traffic on port 443 to port 8889 which is the default port for WebRTC and pointed the domain back to the load balancer. From here you can work on it depending on ur use case as it is ready to publish using the domain.

[a comment by an automated account is not shown]
